test/utils/resolveObject.js

```diff
@@ -119,34 +119,60 @@ describe('resolveObject', function() {
   });
 
   it('should gracefully handle circular references', function() {
-    assert.throws(
-      resolveObject.bind(null,
-        helpers.fileToJSON(__dirname + '/../json_files/circular.json')
-      ),
-      Error,
-      'Circular definition: a | d'
-    );
-    assert.throws(
-      resolveObject.bind(null,
-        helpers.fileToJSON(__dirname + '/../json_files/circular_2.json')
-      ),
-      Error,
-      'Circular definition: a.b.c | d'
-    );
-    assert.throws(
-      resolveObject.bind(null,
-        helpers.fileToJSON(__dirname + '/../json_files/circular_3.json')
-      ),
-      Error,
-      'Circular definition: a.b.c | d.e.f'
-    );
-    assert.throws(
-      resolveObject.bind(null,
-        helpers.fileToJSON(__dirname + '/../json_files/circular_4.json')
-      ),
-      Error,
-      'Circular definition: a.b.c | g.h'
-    );
+    var ERR_TYPE = 'Property Reference Errors';
+    handleErrors.clear(ERR_TYPE);
+
+    resolveObject(helpers.fileToJSON(__dirname + '/../json_files/circular.json'));
+    assert.equal(handleErrors.count(ERR_TYPE), 1);
+    assert.equal(JSON.stringify(handleErrors.fetch(ERR_TYPE)), JSON.stringify([
+       'Circular definition cycle:  a, b, c, d, a'
+    ]));
+    handleErrors.clear(ERR_TYPE);
+
+    resolveObject(helpers.fileToJSON(__dirname + '/../json_files/circular_2.json'));
```

Comment:
We should split these up into different tests

Reply:
Lets do this when we integrate with jest branch